lib/Controller/OpenIdConnectController.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * @author Thomas Müller <thomas.mueller@tmit.eu>
+ * @copyright Copyright (c) 2018, ownCloud GmbH
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ */
+
+namespace OCA\OAuth2\Controller;
+
+use OCP\AppFramework\ApiController;
+use OCP\AppFramework\Http\JSONResponse;
+use OCP\ILogger;
+use OCP\IRequest;
+use OCP\IURLGenerator;
+use OCP\IUserSession;
+
+class OpenIdConnectController extends ApiController {
+
+	/** @var IURLGenerator */
+	private $urlGenerator;
+
+	/** @var ILogger */
+	private $logger;
+
+	/** @var IUserSession */
+	private $userSession;
+
+	/**
+	 * OAuthApiController constructor.
+	 *
+	 * @param string $AppName The app's name.
+	 * @param IRequest $request The request.
+	 * @param IURLGenerator $urlGenerator The URL generator.
+	 * @param ILogger $logger The logger.
+	 */
+	public function __construct($AppName, IRequest $request,
+								IUserSession $userSession,
+								IURLGenerator $urlGenerator,
+								ILogger $logger) {
+		parent::__construct($AppName, $request);
+
+		$this->userSession = $userSession;
+		$this->urlGenerator = $urlGenerator;
+		$this->logger = $logger;
+	}
+
+	/**
+	 * Implements OpenID Connect UserInfo endpoint
+	 *
+	 * @see https://connect2id.com/products/server/docs/api/userinfo
+	 *
+	 * @return JSONResponse The claims as JSON Object.
+	 *
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 * @CORS
+	 * @throws \Exception
+	 */
+	public function userInfo() {
+		$user = $this->userSession->getUser();
+		if ($user === null) {
+			throw new \Exception();
+		}
+		$avatarUrl = $this->urlGenerator->linkTo('', 'remote.php');
+		$avatarUrl .= "/dav/avatars/{$user->getUID()}/96.jpeg";
+
+		$avatarUrl = $this->urlGenerator->getAbsoluteURL($avatarUrl);
+		$data = [
+			"sub"         => $user->getUID(),
+			"picture"     => $avatarUrl
```

Comment:
`"picture"` could also be `null` if the avatar is unset; no? - I think it's better to return null than a URL that resolves into 404. Either that or link the generated avatar (the one with the letters)

So e.g. we can check for null for things like https://github.com/owncloud/oauth2/issues/95 instead of using a broken image link.